File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/activities/MapList.java
package io.github.romangolovanov.apps.ametro.ui.activities;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SearchView;
import androidx.core.app.NavUtils;
import androidx.core.view.MenuItemCompat;

import io.github.romangolovanov.apps.ametro.R;
import io.github.romangolovanov.apps.ametro.app.Constants;
import io.github.romangolovanov.apps.ametro.catalog.entities.MapInfo;
import io.github.romangolovanov.apps.ametro.ui.fragments.MapListFragment;
import io.github.romangolovanov.apps.ametro.ui.loaders.ExtendedMapInfo;

public class MapList extends AppCompatActivity implements MapListFragment.IMapListEventListener{

    private MapListFragment listFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_map_list_view);

        setSupportActionBar(findViewById(R.id.toolbar));
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

        listFragment = (MapListFragment) getSupportFragmentManager().findFragmentById(R.id.list);
        assert listFragment != null;
        listFragment.setMapListEventListener(this);
    }

    @Override
    public void onOpenMap(MapInfo map) {
        Intent viewIntent = new Intent();
        viewIntent.putExtra(Constants.MAP_PATH, map.getFileName());
        setResult(RESULT_OK, viewIntent);
        finish();
    }

    @Override
    public void onLoadedMaps(ExtendedMapInfo[] maps) {
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_map_list, menu);
        MenuItem searchItem = menu.findItem(R.id.action_search);
        SearchView searchView = (SearchView) MenuItemCompat.getActionView(searchItem);
        searchView.setOnQueryTextListener(listFragment);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int itemId = item.getItemId();
        if (itemId == android.R.id.home) {
            NavUtils.navigateUpFromSameTask(this);
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/activities/MapList.kt
package io.github.romangolovanov.apps.ametro.ui.activities

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.SearchView
import androidx.core.app.NavUtils
import io.github.romangolovanov.apps.ametro.R
import io.github.romangolovanov.apps.ametro.app.Constants
import io.github.romangolovanov.apps.ametro.catalog.entities.MapInfo
import io.github.romangolovanov.apps.ametro.ui.fragments.MapListFragment
import io.github.romangolovanov.apps.ametro.ui.loaders.ExtendedMapInfo

class MapList : AppCompatActivity(), MapListFragment.IMapListEventListener {

    private lateinit var listFragment: MapListFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_map_list_view)

        setSupportActionBar(findViewById(R.id.toolbar))
        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)

        listFragment = supportFragmentManager.findFragmentById(R.id.list) as MapListFragment
        listFragment.setMapListEventListener(this)
    }

    override fun onOpenMap(map: MapInfo) {
        val viewIntent = Intent()
        viewIntent.putExtra(Constants.MAP_PATH, map.fileName)
        setResult(RESULT_OK, viewIntent)
        finish()
    }

    override fun onLoadedMaps(maps: Array<ExtendedMapInfo>) {}

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_map_list, menu)
        val searchItem = menu.findItem(R.id.action_search)
        val searchView = searchItem.actionView as SearchView
        searchView.setOnQueryTextListener(listFragment)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            NavUtils.navigateUpFromSameTask(this)
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}
